### DISCUSSION TITLE
Terrain-accurate maps?

### DISCUSSION THREAD

**Discussion by sebkont** [wduda/TravelWindowII#250] [Feature Ideas & Requests]
Hello, so first of all I must be stupid for using the plugin for all this time and literally never before realizing it has map view for each region (which year was that added?). But since it's using the confusing in-game maps where icons cant match geographical locations exactly (like in most other games), and you seem to be actively developing it, it made me wonder - would you be willing to change the map graphics and realign the skills, so it's more intuitive to use? 

I'm asking because I might provide ready-to-go maps, if I make some edits to my PS project files. (If you're planning on ever adding scrollbars, if that's possible - like stable master-map but bigger? - and attaching higher resolution map files, then the final product might be even more modern/impressive/clear.)

**Comment by wduda** [comment]
This is very close to what is being considered in #239. Which maps one would use is not yet decided, but I have researched how I would implement such a feature. Having multiple different map views is also not an issue, just extra effort.

I am not 100% sure I agree with saying the ingame maps are that confusing. For someone very familiar with Middleearth's geography as described in the books or wanting perfect geographical representation that might be the case, but I feel the current maps can be useful because the placement of locations relative to each other kind of works.

But the solution to that would be again offering both kinds of maps as a view, according to the users preference.

**Comment by sebkont** [comment]
As a long-standing LOTRO player, even if I hold that view (to a degree), they are maybe not THAT confusing in practice, yes, but I think they might be to newbies. Specifically the Eriador map has become weirdly cluttered with new additions of landscape and the amount of space they represent, which is just counter-intuitive sometimes (like the vast space between Shire and Evendim but not enough space in other areas). Plus, maybe that's just me (or not?) but finding the right icon, solely by instinct, isn't as quick for me when map proportions/geography differ, I have to use a text label to be sure.

In any case, here is what I meant, with white rectangles quickly placed in Paint to imitate skills 
(I could get rid of white borders/text for actual maps)

Larger, with side bar (if we assume scrollable feature): [https://i.imgur.com/bmTYWzC.jpeg](https://i.imgur.com/bmTYWzC.jpeg )
Smaller, without scroll bar: [https://i.imgur.com/RJYFULS.png](https://i.imgur.com/RJYFULS.png)

Here, you could easily and instantly tell the location of Ost Guruth or Trader's Wharf just by eyeing the color/geographical features of zones, even from a distance. You might have seen it already but it's all viewable online at lotromap.com, [with recently released Harad](https://lotromap.com/harad.html), all aligned based on actual heightmap files, and keeps some of the game's actual style/graphic of maps which was my intention. 

Just a suggestion though, because it's unlikely I'll be able to ever make a fork and implement this idea on my own (plus, slowly but surely, I'm currently busy cooking up my other lotro project long-in-the-making). But if you're ever up for testing this, I could prepare the image.

**Reply by wduda** [reply]
Actually if I make one map based view then making another is just the one-time effort of setting the coordinates where the icons should be on the map. And whiterabbit built some tooling for finding the coordinates already. 

So I am very willing to commit to having both views - ingame and bespoke - we would just need to think about the right resolution and the scrollability. My thoughts are that if I am leveling in the Isengard range and have to scroll every time I want to go to Galtrev that would be a bad experience. So maybe we would save the scrolled to position every time and open back to that, or prefer a non-scrollable version. 

Would you be able to provide your images without the white squares? Just Eriador for now is enough to test what it would look like in the plugin.

**Reply by sebkont** [reply]
Oh, yeah, saving the scroll position might be a good solution, if that's possible. Currently the map is [tiny like this](https://i.imgur.com/rLgceOH.jpeg) on my monitor, for example, so having a bit larger view with higher res file that's scrollable could be a good user experience.

Here is the file: [https://www.file.io/MLba/download/bUGgAvpWKDRH](https://www.file.io/MLba/download/bUGgAvpWKDRH)
I left the zone boundaries for now. Depending on how it looks and the method/resolution you choose, I may modify as needed or maybe add the names of zones back.  

Also, in the plugin's future, do you plan on keeping the current map divide? Not like the 'system regions' (separate big maps) of SSG but Rohan separately from Rhovanion, all of Gondor (siege and After Battle merged into one, with double locations = before and after Osgiliath, for example). Since on the maps I made, I split them like SSG has done it internally, and categorized their "Gondor" map as "Gondor Besieged" and their "Mordor" map as "After Battle" (with After Battle version of Gondor and Mordor). Which I feel is more intuitive navigation rather than trying to distinguish between Battle/After Minas Tirith icons on a single map. But if needed, it shouldn't be too hard for me to make edits and do the 'joint' visual version of Gondor (or split Rohan), like you have it right now.  So whatever you like best

**Reply by wduda** [reply]
There was an error retrieving the file. Can you upload it again maybe? 
Leaving zone boundaries is fine I would be building a prototype for now.

**Reply by sebkont** [reply]
> There was an error retrieving the file. Can you upload it again maybe? Leaving zone boundaries is fine I would be building a prototype for now.

Ok, here: [https://easyupload.io/r3m8yc](https://easyupload.io/r3m8yc )

**Reply by wduda** [reply]
Thanks. downloaded a few days ago.

**Comment by wduda** [comment]
Comment by Cay in https://forums.lotro.com/index.php?threads/plugin-travel-window-ii.1400/post-186936:

Ok, one more spam here. Last, I promise. :geek:

I thought a bit how could this get implemented easily but also effectively and came up with this:

- One button (preferably the suitcase already there and you could then also opt to use the map instead the other travel menues) that opens the map you last viewed. This it to avoid the menu hassle.

- Implementing transition buttons exactly like the ingame maps have where you can click them to quickly swap which map is shown, without the clunky map menu.

I think these two would be sufficient to perfect the map side of your plugin and you could then avoid doing any of those time consuming zoom/scroll/drag things.

**Comment by wduda** [comment]
@sebkont this feature is now in a status where it is workable, but uses the existing maps. I want to offer an alternative version soon which will use the images you provided. For a full feature one would need more than one map image though to cover Eriador, Mordor, and Harad at least.

**Reply by wduda** [reply]
If you know how, the code with the feature is available in the map view branch - see also https://github.com/wduda/TravelWindowII/pull/266

**Comment by wduda** [comment]
The feature has now been there quite some time and is stable, with some added polish over time. I will close the discussion for now.
